app/controllers/content_items_controller.rb

```diff
@@ -24,6 +24,35 @@ class ContentItemsController < ApplicationController
   def show
     load_content_item
 
+    # TEMPORARY (author: richard.towers, expected end date: November 30 2023)
+    # Content specific AB test for the Find your UTR number page
+    if @content_item.base_path == "/find-utr-number"
+      ab_test = GovukAbTesting::AbTest.new(
+        "find_utr_number_video_links",
+        dimension: 300, # TODO: which dimension should we use?
+        allowed_variants: %w[HelpText VideoLink],
+        control_variant: "HelpText",
+      )
+      @requested_variant = ab_test.requested_variant(request.headers)
+      @requested_variant.configure_response(response)
+
+      if @requested_variant.variant? "VideoLink"
+        # NOTE: this is a fragile way of doing an AB test on content.
+        #
+        # Any change to the base content, or to the way the content is rendered
+        # could potentially break the B variant of the test, and result in both
+        # variants being the same.
+        # We're aware of this risk, and we're going to be careful in this one off
+        # situation. This is not a sustainable way of doing AB tests in the
+        # future.
+        @content_item.body.sub!(
```

Comment:
As you've noted, if HMRC update the text in line 49 things break. We would still be recording variant "VideoLink" but clicks to the video would be reduced (because the video isn't there) - ie our data would be polluted. So I was wondering if there's a way to reduce that/bring it back under our control? Could we read the variant A and variant B text from a yaml file in this application? We could put the logic to fetch that data in the ContentItem::Body module maybe. And then once this AB test logic is live, we could re_publish the content item with an obvious "AB_TEST_MATCHING_STRING_DO_NOT_REMOVE" kind of thing. And then if they want to update either variant, they'd have to make a request to GDS to update it for them. It's still a hack! But maybe a more obvious and visible one?